File: Algo/Leetcode/448AllNumDisappearInArray.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for i, n in enumerate(nums):
            nums[abs(n) - 1] = - abs(nums[abs(n) - 1])

        res = []
        for i, n in enumerate(nums):
            if n > 0:
                res.append(i+1)

        return res

File: Algo/Leetcode/448AllNumDisappearInArray.cpp
// O(n) time, O(1) extra space
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        for(int i=0; i<nums.size(); i++)
            nums[abs(nums[i]) - 1] = - abs(nums[abs(nums[i]) - 1]); // must set to negative

        vector<int> res;
        for(int i=0; i<nums.size(); i++)
            if(nums[i] > 0) res.push_back(i+1);

        return res;
    }
};